File: wearbyApp/src/main/java/com/wearby/sesion/SesionUsuario.java
package com.wearby.sesion;

/**
 * Almacena los datos del usuario autenticado durante la sesión.
 * Patrón Singleton: Una única estancia accesible globalmente
 */
public class SesionUsuario {

    private static SesionUsuario instancia;

    private Integer id;
    private String nombre;
    private String email;
    private String rol;

    private SesionUsuario(){}

    public static SesionUsuario getInstancia() {
        if (instancia == null){
            instancia = new SesionUsuario();
        }
        return instancia;
    }

    public void iniciar(Integer id, String nombre, String email, String rol) {
        this.id = id;
        this.nombre = nombre;
        this.email = email;
        this.rol = rol;
    }

    public void cerrar() {
        this.id = null;
        this.nombre = null;
        this.email = null;
        this.rol = rol;
    }

    public boolean isAdmin(){
        return "admin".equals(rol);
    }

    public Integer getId() {return id;}
    public String getNombre() {return nombre;}
    public String getEmail() {return email;}
    public String getRol() {return rol;}
}

File: werarbyAndroid/app/src/main/java/com/example/wearbyandroid/sesion/SesionUsuario.kt
package com.example.wearbyandroid.sesion

import com.example.wearbyandroid.modelo.Usuario

/**
 * Almacena los datos del usuario autenticado durante la sesión.
 * Patrón Singleton: una única instancia accesible globalmente.
 */
object SesionUsuario {

    private var usuario: Usuario? = null

    fun iniciar(u: Usuario) {
        usuario = u
    }

    fun cerrar() {
        usuario = null
    }

    fun getId(): Int = usuario?.id ?: 0
    fun getNombre(): String = usuario?.nombre ?: ""
    fun getEmail(): String = usuario?.email ?: ""
    fun getRol(): String = usuario?.rol ?: ""
    fun isAdmin(): Boolean = usuario?.rol == "admin"
    fun isLoggedIn(): Boolean = usuario != null
}